src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -454,23 +489,28 @@ App::DocumentObject* AssemblyObject::getJointOfPartConnectingToGround(App::Docum
     return nullptr;
 }
 
-JointGroup* AssemblyObject::getJointGroup()
+JointGroup* AssemblyObject::getJointGroup(App::Part* part)
 {
-    App::Document* doc = getDocument();
+    App::Document* doc = part->getDocument();
 
     std::vector<DocumentObject*> jointGroups =
         doc->getObjectsOfType(Assembly::JointGroup::getClassTypeId());
     if (jointGroups.empty()) {
         return nullptr;
     }
     for (auto jointGroup : jointGroups) {
-        if (hasObject(jointGroup)) {
+        if (part->hasObject(jointGroup)) {
             return dynamic_cast<JointGroup*>(jointGroup);
         }
     }
     return nullptr;
 }
 
+JointGroup* AssemblyObject::getJointGroup()
```

Comment:
If possible declare it as const. Therefore the argument of getJointGroup() must be const, too.